sympy/utilities/tests/test_code_quality.py

```diff
@@ -37,6 +38,51 @@
 message_self_assignments = "File contains assignments to self/cls: %s, line %s."
 message_func_is = "File contains '.func is': %s, line %s."
 
+# XXX Python 2 unicode import test.
+# May remove after deprecating python 2.7.
+message_py2_unicode_not_whitelisted = \
+    "File contains a unicode character : %s, line %s. " \
+    "But with no encoding header. " \
+    "See https://www.python.org/dev/peps/pep-0263/ " \
+    "and add '# coding=utf-8'"
+message_py2_unicode_whitelisted = \
+    "File contains a unicode character : %s, line %s. " \
+    "But not in the whitelist " \
+    "If necessary, add the file to the whitelist in " \
+    "'test_code_quality.py'"
+message_unnecessary_unicode_cookie = \
+    "File contains a unicode encoding header : %s, line %s. " \
+    "But it would not be needed."
+
+encoding_header_re = re.compile(
+    r'^[ \t\f]*#.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)')
+
+# Whitelist pattern for files which can have unicode.
+unicode_whitelist = [
+    r'*/bin/authors_update.py',
+    r'*/bin/mailmap_update.py',
+    r'*/physics/quantum/tests/test_printing.py',
+    r'*/sympy/physics/vector/tests/test_printing.py',
+    r'*/sympy/vector/tests/test_printing.py',
+    r'*/parsing/*',
+    r'*/printing/*',
+    # Ignore junk files
+    r'*/.ipynb_checkpoints/*',
+
+    # TODO Delete unicode in these files.
+    r'*/combinatorics/free_groups.py',
+    r'*/combinatorics/tests/test_coset_table.py',
+    r'*/crypto/crypto.py',
+    r'*/discrete/transforms.py',
+    r'*/liealgebras/root_system.py',
+    r'*/liealgebras/type_g.py',
+    r'*/liealgebras/weyl_group.py',
+    r'*/liealgebras/tests/test_type_G.py',
+    r'*/ntheory/residue_ntheory.py',
+    r'*/physics/mechanics/rigidbody.py',
+    r'*/polys/subresultants_qq_zz.py'
```

Comment:
Oh I see these are the files. This file in particular seems to be using curly quotes. Those could be replaced with regular quotes (it's up to the doc renderer to render curly quotes for typography). 

discrete/transforms and residue_ntheory use "Möbius" in some of the docstrings. That seems fine to me. 

The liealgebra files use greek symbols in docstrings, which could be replaced with latex. 

rigitbody.py uses `H = I·w + r×Mv`, which could be redone in latex.

crypto.py has some en-dashes, which could be replaced with regular dashes, and "Vigenère", which is fine.

free_groups.py has an en-dash which should clearly be a regular dash (it's in a LaTeX equation)

test_coset_table.py has some subscript numbers in comments (like `₁`), which could be replaced with normal numbers